File: Pycharm Projects/GettingStarted/getting_started.py
""" -----------------------------------------------------------------------
 * <copyright file="main.py" company="Hybrid Software Helix Ltd">
 *  Copyright (C) 2025 Hybrid Software Helix Ltd. All rights reserved.
 * </copyright>
 * <summary>
 *  This example is provided on an "as is" basis and without warranty of any kind.
 *  Hybrid Software Helix Ltd. does not warrant or make any representations
 *  regarding the use or results of use of this example.
 * </summary>
 * -----------------------------------------------------------------------
 """

import os
from jawsmakoIF_python import *

# Instantiate Mako
jawsmako = IJawsMako.create("")
IJawsMako.enableAllFeatures(jawsmako)
factory = jawsmako.getFactory()

# Create an assembly, document and page
assembly = IDocumentAssembly.create(jawsmako)
document = IDocument.create(jawsmako)
assembly.appendDocument(document)
page = IPage.create(jawsmako)
document.appendPage(page)

# Create a fixed page A5 landscape.
pagewidth = 793.7
pageheight = 561.1
fixedpage = IDOMFixedPage.create(factory, pagewidth, pageheight)

# Now create some DOM for the page contents.

# Find a font
font: IDOMFont = jawsmako.findFont('Arial Bold')

# Create a blue brush
hcolorspace = IDOMColorSpacesRGB.create(factory)
bluecolor = IDOMColor.create(factory, hcolorspace, 1.0, 0.0, 0.0, 1.0)
solidBrush = IDOMSolidColorBrush.create(factory, bluecolor)

# Create glyphs
message = 'Mako Python API'
glyphs = IDOMGlyphs.create(factory, message, 72, FPoint(10, pageheight / 2),
                           solidBrush, font[0], 0)

# And add to the page
fixedpage.appendChild(glyphs.toIDOMNode())

# Assign content to a page
page.setContent(fixedpage)

# Now we can write this to a PDF
dir_path = os.path.dirname(os.path.realpath(__file__))
IPDFOutput.create(jawsmako).writeAssembly(assembly, dir_path + '/TestPython.pdf')

print('File created.')

File: IntelliJ Projects/GettingStarted/src/GettingStarted.java
/* -----------------------------------------------------------------------
 * <copyright file="Main.java" company="Hybrid Software Helix Ltd">
 *  Copyright (C) 2025 Hybrid Software Helix Ltd. All rights reserved.
 * </copyright>
 * <summary>
 *  This example is provided on an "as is" basis and without warranty of any kind.
 *  Hybrid Software Helix Ltd. does not warrant or make any representations
 *  regarding the use or results of use of this example.
 * </summary>
 * -----------------------------------------------------------------------
 */

import com.globalgraphics.JawsMako.jawsmakoIF.*;

public class GettingStarted {
    public static void main(String[] args) {

        var jawsMako = IJawsMako.create();
        var factory = jawsMako.getFactory();
        IJawsMako.enableAllFeatures(jawsMako);
        var assembly = IDocumentAssembly.create(jawsMako);
        var document = IDocument.create(jawsMako);
        assembly.appendDocument(document);
        var page = IPage.create(jawsMako);
        document.appendPage(page);

        // Create a fixed page A5 landscape.
        double pageWidth = 793.7;
        double pageHeight = 561.1;
        var fixedPage = IDOMFixedPage.create(factory, pageWidth, pageHeight);

        // Now create some DOM for the page contents.

        // Find a font
        long[] fontIndex = {0};
        IDOMFont font = jawsMako.findFont("Arial Bold", fontIndex);

        // Create a blue brush
        var colorSpace = IDOMColorSpacesRGB.create (factory);
        var colorValues = new StdVectFloat();
        colorValues.add(0.0f);
        colorValues.add(0.0f);
        colorValues.add(1.0f);

        var blueColor = IDOMColor.createFromVect(factory, colorSpace, 1.0f, colorValues);
        var solidBrush = IDOMSolidColorBrush.create(factory, blueColor);

        // Create glyphs
        String message = "Mako Java API";
        var glyphs = IDOMGlyphs.create(factory, message, 72, new FPoint(10, pageHeight / 2), solidBrush, font, fontIndex[0]);

        // And add to the page
        fixedPage.appendChild(glyphs.toIDOMNode());


        // Assign content to a page
        page.setContent(fixedPage);

        // Now we can write this to a PDF
        IPDFOutput.create(jawsMako).writeAssembly(assembly, "TestJava.pdf");

        System.out.println("File created.");
    }
}